Caching test fails unittest test suite

For whatever reason, test_caching of multifeaturizer fails the test suite now that we transitioned to unittests from old (deprecated) nose. 

What is strange is that the test seems to pass locally by using `python test_base.py` (i.e., running `unittest.main()`). 

Locally, running `python setup.py test` fails with:
```bash
======================================================================
FAIL: test_caching (matminer.featurizers.tests.test_base.TestBaseClass)
Test whether MultiFeaturizer properly caches
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/ardunn/alex/lbl/projects/common_env/dev_codes/matminer/matminer/matminer/featurizers/tests/test_base.py", line 416, in test_caching
    self.assertEqual(2, _get_all_nearest_neighbors.cache_info().hits)
AssertionError: 2 != 15
----------------------------------------------------------------------
```

On circleCI though, the test has this similar failure:

```bash
======================================================================
FAIL: test_caching (matminer.featurizers.tests.test_base.TestBaseClass)
Test whether MultiFeaturizer properly caches
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/circleci/matminer/matminer/featurizers/tests/test_base.py", line 416, in test_caching
    self.assertEqual(2, _get_all_nearest_neighbors.cache_info().hits)
AssertionError: 2 != 8
```

So the problem only seems to happen when running all the tests. 

Perhaps @WardLT knows where to start on a fix?

P.S. Currently this test is being skipped on CircleCI

--- a/matminer/featurizers/tests/test_base.py
+++ b/matminer/featurizers/tests/test_base.py
@@ -403,6 +403,9 @@ class TestBaseClass(PymatgenTest):
             SiteStatsFingerprint.from_preset("LocalPropertyDifference_ward-prb-2017"),
             SiteStatsFingerprint.from_preset("CoordinationNumber_ward-prb-2017")
         ], iterate_over_entries=True)
+        
+        # Reset the cache before tests
+        _get_all_nearest_neighbors.cache_clear()
 
         # Create a dataframe with two SC structures in it
         data = pd.DataFrame({'strcs': [

